backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/services/GeneralViewNodeToolSectionSwitch.java

```diff
@@ -60,83 +62,130 @@ public class GeneralViewNodeToolSectionSwitch extends SysmlEClassSwitch<Void> {
+        return List.of(createSection, this.addElementsToolSection());
     }
 
     @Override
-    public Void caseRequirementDefinition(RequirementDefinition object) {
-        this.nodeToolSections.add(this.createPartUsageAsRequirementSubject());
-        return super.caseRequirementDefinition(object);
+    public List<NodeToolSection> caseRequirementDefinition(RequirementDefinition object) {
+        var createSection = this.buildCreateSection(this.createPartUsageAsRequirementSubjectNodeTool());
+        createSection.getNodeTools().addAll(this.createToolsForCompartmentItems(object));
+        return List.of(createSection, this.addElementsToolSection());
     }
 
     @Override
-    public Void caseUsage(Usage object) {
-        this.createToolsForCompartmentItems(object);
-        return super.caseUsage(object);
+    public List<NodeToolSection> caseUsage(Usage object) {
+        var createSection = this.buildCreateSection();
+        createSection.getNodeTools().addAll(this.createToolsForCompartmentItems(object));
+        return List.of(createSection, this.addElementsToolSection());
     }
 
-    private NodeToolSection createPartUsageAsRequirementSubject() {
+    private NodeTool createPartUsageAsRequirementSubjectNodeTool() {
         var serviceCall = this.viewBuilderHelper.newChangeContext().expression("aql:self.createRequirementSubject(self.eContainer().eContainer())");
-        var createSubjectTool = this.diagramBuilderHelper.newNodeTool()
+        return this.diagramBuilderHelper.newNodeTool()
                 .name("New Subject")
                 .iconURLsExpression("/icons/full/obj16/Subject.svg")
                 .preconditionExpression("aql:self.isEmptySubjectCompartment()")
                 .body(serviceCall.build()).build();
-        return this.diagramBuilderHelper.newNodeToolSection()
-                .name("Create")
-                .nodeTools(createSubjectTool)
-                .build();
     }
 
-    private NodeToolSection createPartDefinitionElementsToolSection(NodeDescription partUsageNodeDescription, NodeDescription itemUsageNodeDescription) {
-        return this.diagramBuilderHelper.newNodeToolSection()
-                .name("Create")
-                .nodeTools(this.createNestedUsageNodeTool(partUsageNodeDescription, SysmlPackage.eINSTANCE.getPartUsage()),
-                        this.createNestedUsageNodeTool(itemUsageNodeDescription, SysmlPackage.eINSTANCE.getItemUsage()))
-                .build();
+    private NodeToolSection createPartDefinitionElementsToolSection() {
+        return this.buildCreateSection(
+                this.createNestedUsageNodeTool(SysmlPackage.eINSTANCE.getPartUsage()),
+                this.createNestedUsageNodeTool(SysmlPackage.eINSTANCE.getItemUsage()));
     }
 
     private NodeToolSection createPartUsageElementsToolSection(NodeDescription nodeDesc) {
-        return this.diagramBuilderHelper.newNodeToolSection()
```

Comment:
` private NodeToolSection createPartUsageElementsToolSection(NodeDescription nodeDesc)` the parameter `nodeDesc` seems not used anymore. Could you remove it please?

Reply:
Donc and the NodeDescription field is no longer needed as well.